sync/syncer.go

```diff
@@ -76,41 +78,40 @@ func (s *Syncer) Start() {
 
 //fires a sync every sm.syncInterval or on force space from outside
 func (s *Syncer) run() {
-	syncTicker := time.NewTicker(s.syncInterval)
+	foo := func() {
+		if atomic.CompareAndSwapUint32(&s.SyncLock, IDLE, RUNNING) {
+			s.Synchronise()
+			atomic.StoreUint32(&s.SyncLock, IDLE)
+		}
+	}
 	for {
-		doSync := false
 		select {
 		case <-s.exit:
-			log.Debug("run stoped")
+			s.Debug("run stoped")
 			return
-		case doSync = <-s.forceSync:
-		case <-syncTicker.C:
-			doSync = true
-		}
-		if doSync {
-			go func() {
-				if atomic.CompareAndSwapUint32(&s.SyncLock, IDLE, RUNNING) {
-					s.Debug("do sync")
-					s.Synchronise()
-					atomic.StoreUint32(&s.SyncLock, IDLE)
-				}
-			}()
+		case <-s.forceSync:
+			go foo()
+		case layer := <-s.clock:
+			atomic.StoreUint32(&s.currentLayer, uint32(layer))
+			s.Info("sync got tick for layer %v", layer)
+			go foo()
```

Comment:
Just to make sure I understand - this is where layers are pushed to the Tortoise, right? 
If so, why do you wait till the beginning of the layer in order to push the previous layer to the tortoise? You can do that after a minute (or less) from the beginning of the previous layer when all the honest blocks had arrived. 
Another "just to make sure" - this doesn't handle late blocks, right? In the current state of the code late blocks will be ignored?

Reply:
you got it right, late block handling will be add with #565 
and the concurrent handling will also be a separate pr